.buildkite/scripts/rewrite.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+if [[ -z "${BUILDKITE_PULL_REQUEST:-}" ]]; then
+  echo "Not a pull request, skipping rewrite"
+  exit 0
+fi
+
+if ! git diff --exit-code; then
+  echo "Changes are present before running rewrite, not running."
+  git status
+  exit 0
+fi
+
+NEW_COMMIT_MESSAGE="[CI] Auto commit changes from rewrite"
+PREVIOUS_COMMIT_MESSAGE="$(git log -1 --pretty=%B)"
+
+echo "--- Running rewrite"
+.ci/scripts/run-gradle.sh -Dscan.tag.NESTED rewriteRun
+
+if git diff --exit-code; then
+  echo "No changes found after running rewrite. Don't need to auto commit."
+  exit 0
+fi
+
+if [[ "$NEW_COMMIT_MESSAGE" == "$PREVIOUS_COMMIT_MESSAGE" ]]; then
+  echo "Changes found after running rewrite."
+  echo "CI already attempted to commit these changes, but the file(s) seem to have changed again."
+  echo "Please review and fix manually."
+  exit 1
+fi
+
+git config --global user.name elasticsearchmachine
+git config --global user.email 'infra-root+elasticsearchmachine@elastic.co'
+
+gh pr checkout "${BUILDKITE_PULL_REQUEST}"
+git add -u .
+git commit -m "$NEW_COMMIT_MESSAGE"
```

Comment:
@romani we could do like this in the next lvl of integration:

- https://github.com/checkstyle/checkstyle/pull/17545